src/Database/exceptions.php

```diff
@@ -45,4 +45,43 @@ class NotNullConstraintViolationException extends ConstraintViolationException
  */
 class UniqueConstraintViolationException extends ConstraintViolationException
 {
+	/** @var string */
+	private $entry;
+
+	/** @var string */
+	private $key;
+
+
+	/**
+	 * @returns self
+	 */
+	public static function from(\PDOException $src)
+	{
+		$e = parent::from($src);
+
+		preg_match("/Duplicate entry '([^']+)' for key '([^']+)'/", $e->errorInfo[2], $m);
```

Comment:
This needs checks whether it matched anything or not (that's why Travis fails).
Also this check is not platform-independent. Have a look at Doctrine/DBAL how they handle this: https://github.com/doctrine/dbal/search?utf8=%E2%9C%93&q=UniqueConstraintViolationException